laboratory-service/src/main/java/com/raf/si/laboratoryservice/services/impl/WorkOrderServiceImplementation.java

```diff
@@ -0,0 +1,196 @@
+package com.raf.si.laboratoryservice.services.impl;
+
+import antlr.Token;
+import com.raf.si.laboratoryservice.dto.request.order.*;
+import com.raf.si.laboratoryservice.dto.response.order.*;
+import com.raf.si.laboratoryservice.exception.BadRequestException;
+import com.raf.si.laboratoryservice.exception.NotFoundException;
+import com.raf.si.laboratoryservice.mapper.OrderMapper;
+import com.raf.si.laboratoryservice.model.*;
+import com.raf.si.laboratoryservice.model.enums.labworkorder.OrderStatus;
+import com.raf.si.laboratoryservice.model.enums.referral.ReferralStatus;
+import com.raf.si.laboratoryservice.model.enums.user.Profession;
+import com.raf.si.laboratoryservice.model.enums.user.Title;
+import com.raf.si.laboratoryservice.repository.*;
+import com.raf.si.laboratoryservice.services.WorkOrderService;
+import com.raf.si.laboratoryservice.utils.TokenPayload;
+import com.raf.si.laboratoryservice.utils.TokenPayloadUtil;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.criterion.Order;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.parameters.P;
+import org.springframework.stereotype.Service;
+import java.util.*;
+
+@Slf4j
+@Service
+public class WorkOrderServiceImplementation implements WorkOrderService {
+
+    ReferralRepository referralRepository;
+    LabAnalysisRepository labAnalysisRepository;
+    LabWorkOrderRepository labWorkOrderRepository;
+    AnalysisParameterResultRepository analysisParameterResultRepository;
+    OrderMapper orderMapper;
+    AnalysisParameterRepository analysisParameterRepository;
+
+    public WorkOrderServiceImplementation(ReferralRepository referralRepository, LabAnalysisRepository labAnalysisRepository, LabWorkOrderRepository labWorkOrderRepository, AnalysisParameterResultRepository analysisParameterResultRepository, OrderMapper orderMapper, AnalysisParameterRepository analysisParameterRepository) {
+        this.referralRepository = referralRepository;
+        this.labAnalysisRepository = labAnalysisRepository;
+        this.labWorkOrderRepository = labWorkOrderRepository;
+        this.analysisParameterResultRepository = analysisParameterResultRepository;
+        this.orderMapper = orderMapper;
+        this.analysisParameterRepository = analysisParameterRepository;
+    }
+
+    @Override
+    public OrderResponse createOrder(Long orderId) {
+        TokenPayload tokenPayload = TokenPayloadUtil.getTokenPayload();
+        UUID lbz = tokenPayload.getLbz();
+
+        Optional<Referral> referral = referralRepository.findById(orderId);
+        if(referral.isEmpty()){
+            String errMessage = String.format("Uput sa id-om '%s' ne postoji", orderId);
+            log.info(errMessage);
+            throw new BadRequestException(errMessage);
+        }
```

Comment:
Zasto ne iskoristimo potencijal Optional klase i odradimo referralRepository.findById(orderId).orElseThrow(() -> { log and throw exception});